File: android-jsBridge/src/main/java/me/yamlee/jsbridge/jscall/OpenUriProcessor.java
package me.yamlee.jsbridge.jscall;

import android.content.Context;


import org.json.JSONException;
import org.json.JSONObject;

import me.yamlee.jsbridge.BaseJsCallProcessor;
import me.yamlee.jsbridge.JsCallData;
import me.yamlee.jsbridge.NativeComponentProvider;
import me.yamlee.jsbridge.ui.NearInteraction;

/**
 * 根据uri做指定的跳转动作，比如Uri是"nearmcht://view-login"就会跳转到登录界面
 *
 * @author yamlee
 */
public class OpenUriProcessor extends BaseJsCallProcessor {

    public static final String FUNC_NAME = "openUri";

    public OpenUriProcessor(NativeComponentProvider componentProvider) {
        super(componentProvider);
    }

    @Override
    public boolean onHandleJsQuest(JsCallData callData) {
        if (FUNC_NAME.equals(callData.getFunc())) {
            JSONObject jsonObject = null;
            try {
                jsonObject = new JSONObject(callData.getParams());
                String uri = (String) jsonObject.get("uri");
                NearInteraction interaction = componentProvider.provideWebInteraction();
                Context context = componentProvider.provideApplicationContext();
                return true;
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }

        return false;
    }

    @Override
    public String getFuncName() {
        return FUNC_NAME;
    }
}

File: android-jsBridgeCore/src/main/java/me/yamlee/jsbridge/jscall/OpenUriProcessor.kt
package me.yamlee.jsbridge.jscall


import me.yamlee.jsbridge.BaseJsCallProcessor
import me.yamlee.jsbridge.JsCallData
import me.yamlee.jsbridge.NativeComponentProvider
import me.yamlee.jsbridge.JsCallback
import org.json.JSONException
import org.json.JSONObject

/**
 * Activity will go to the activity which the uri target
 * @author yamlee
 */
class OpenUriProcessor(componentProvider: NativeComponentProvider) : BaseJsCallProcessor(componentProvider) {

    override fun onHandleJsQuest(callData: JsCallData, callback: JsCallback): Boolean {
        if (FUNC_NAME == callData.func) {
            var jsonObject: JSONObject? = null
            try {
                jsonObject = JSONObject(callData.params)
                val uri = jsonObject.get("uri") as String
                val interaction = componentProvider.provideWebInteraction()
                val context = componentProvider.provideApplicationContext()
                return true
            } catch (e: JSONException) {
                e.printStackTrace()
            }

        }

        return false
    }

    override fun getFuncName(): String {
        return FUNC_NAME
    }

    companion object {

        const val FUNC_NAME = "openUri"
    }
}
